src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_capability_updated_notification.cc

```diff
@@ -157,6 +157,17 @@ void OnSystemCapabilityUpdatedNotification::Run() {
       [&system_capability_type,
        &initial_connection_key](const ApplicationSharedPtr app) {
         DCHECK_OR_RETURN(app, false);
+        bool is_registered = false;
+        {
+          auto accessor = app->get_registration_status_accessor();
+          is_registered = Application::ApplicationRegisterState::kRegistered ==
+                          accessor.GetData();
+        }
+        if (!is_registered) {
+          LOG4CXX_WARN(logger_,
+                       "App " << app->app_id() << " is not registered");
+          return false;
+        }
```

Comment:
@SNiukalov Think about the following approach: 
```cpp
{
  auto accessor = app->get_registration_status_accessor();
  const bool is_registered = Application::ApplicationRegisterState::kRegistered ==
                          accessor.GetData();
  if (!is_registered) {
    LOG4CXX_WARN(logger_, "App " << app->app_id() << " is not registered");
    return false;
  }
}
```

ìs_registered declared & defined only where it is needed. The code is more shore & clear.

Reply:
@AByzhynar 
I agree, this is a shorter and more understandable code.
But in this case, we lock the mutex and must release it as soon as possible.
Therefore, I would not want to include in the lock, what can be excluded and does not require synchronization.

Reply:
@SNiukalov You perform the entire locking to get the actual status and write it to `is_registered` variable. Then you release the lock and checking the status outside of the lock while the actual status can be already changed. So perform the check inside the lock in order to guarantee that no other threads can change app status.

Reply:
@AByzhynar Seems it doesn't matter as the status could change right after the release anyway.
Instead I'd suggest

```cpp
const auto is_registered =
    Application::ApplicationRegisterState::kRegistered ==
    app->registration_status_accessor().GetData();
if (!is_registered) {
  LOG4CXX_WARN(logger_,
               "App " << app->app_id() << " is not registered");
  return false;
}
```

Reply:
@mvorobio 
- That's why I proposed to check the status inside lock
- I can't see any reason to change bool to auto type.

Reply:
@AByzhynar No problems with `auto`.
I meant that here is only synchronization of access to the app status to read its value.